src/models/CDD.java

```diff
@@ -772,6 +772,15 @@ public CDD transitionBack(CDD guard, CDD update, int[] clockResets, int[] boolRe
         checkForNull();
         guard.checkForNull();
         update.checkForNull();
+
+        if (isTrue() || isFalse()) {
+            return this;
+        }
+
+        if (update.isTrue() || update.isFalse()) {
+            return this;
+        }
+
         return new CDD(CDDLib.transitionBack(pointer, guard.pointer, update.pointer, clockResets, boolResets)).removeNegative().reduce();
```

Comment:
This is a very good point. Yes, I am pretty sure it would still work as the pointer will always represent the actual CDD value. Only the "Java" side of things can be "dirty".